components/script/dom/webglprogram.rs

```diff
@@ -27,18 +28,142 @@ use crate::dom::webglshader::WebGLShader;
 use crate::dom::webgluniformlocation::WebGLUniformLocation;
 use crate::script_runtime::CanGc;
 
-#[dom_struct]
-pub(crate) struct WebGLProgram {
-    webgl_object: WebGLObject,
+#[derive(JSTraceable, MallocSizeOf)]
+struct DroppableWebGLProgram {
     #[no_trace]
     id: WebGLProgramId,
     is_in_use: Cell<bool>,
     marked_for_deletion: Cell<bool>,
+    fragment_shader: MutNullableDom<WebGLShader>,
```

Comment:
Any Dom value like this is JS-owned and unsafe to use from the destructor. We'll need to look more closely at this.

Reply:
Could the solution be creating a WeakNullableRef struct?

Reply:
Does `RefCell<Option<WeakRef<WebGLShader>>>` work? Sorry for missing this question; I was reminded of this PR by #40655.

Reply:
I'll try